File: fixtures/scrapy_output/put_with_T_option.py
from scrapy import Request, Spider


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        headers = {
            'Content-Type': 'application/json',
        }
        body = '{"properties": {"email": {"type": "keyword"}}}'
        yield Request(
            'http://localhost:9200/twitter/_mapping/user?pretty',
            callback=self.parse,
            method='PUT',
            headers=headers,
            body=body,
        )

    def parse(self, response):
        pass

File: test/fixtures/node-got/put_with_T_option.js
import got from 'got';

const response = await got.put('http://localhost:28139/twitter/_mapping/user?pretty', {
  headers: {
    'Content-Type': 'application/json'
  },
  // body: '{"properties": {"email": {"type": "keyword"}}}',
  json: {
    'properties': {
      'email': {
        'type': 'keyword'
      }
    }
  }
});
